### DISCUSSION TITLE
[Question] Why do we have a SemiTransactionalHiveMetastore in the Hive Connector?

### DISCUSSION THREAD

**Discussion by dingqiaoyi** [trinodb/trino#15025] [Q&A]
Why do we name the HiveMetastore with a prefix 'Semi' as SemiTransactionalHiveMetastore ?
Is it because that we have the Hive connector with partial transactional support, for example lacking in concurrency control, although atomicity seems to be satisfied ?

**Comment by dingqiaoyi** [comment]
@martint Do you know the history?

**Reply by hashhar** [reply]
Because it tries to simulate transactions by maintaining a list of actions and compensation actions to be performed. Each action may require different metastore operation so transactionality cannot be guaranteed hence it's "Semi-tramsactional".
